pkgs/top-level/python-packages.nix

```diff
@@ -11890,6 +11890,20 @@ in modules // {
     };
   });
 
+  libgpuarray-cuda = callPackage ../development/python-modules/libgpuarray/cuda/default.nix rec {
+    numpy = self.numpy_1_10;
+    scipy = self.buildScipyPackage rec {
+      version = "0.17.1";
+      src = pkgs.fetchurl {
```

Comment:
The override mentioned of the `numpy` version in `scipy` turns out to be a little more tricky. If you put
```nix
{
numpy = self.numpy_1_10;
scipy = self.scipy.override { numpy = self.numpy_1_10; };
}
```
in the definition of `libgpuarray`, the nix-builder is going to complain that it has found duplicated packages in closure for dependency 'numpy'. `scipy` is constructed in terms of `buildScipyPackage`, which takes two arguments. It is then followed by `buildPythonPackage`, which invokes `mkDerivation`. If I understand correctly, to achieve the required effect with `numpy`, one should override the attributes in the final set which have made use of `numpy`. Here is an override which seems to work:
```nix
{
numpy = self.numpy_1_10;
scipy = self.scipy.override (let inherit numpy; in {
  passthru = { blas = numpy.blas; };
  propagatedBuildInputs = [ numpy.blas numpy ] ++ [ python pkgs.setuptools ];
});
}
```